install_db4.sh configure cannot guess build type on ppc64le

# Describe the issue

The configure script called by `install_db4.sh` fails to detect the build arch on ppc64le systems, producing an error.

# What behavior did you expect?

db4 should build properly without errors.

# What was the actual behavior (provide screenshots if the issue is GUI-related)?

~~~
checking build system type... /home/user/Downloads/bitcoin/db4/db-4.8.30.NC/dist/config.guess: unable to guess system type

This script, last modified 2009-02-03, has failed to recognize
the operating system you are using. It is advised that you
download the most up to date version of the config scripts from

  http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.guess;hb=HEAD
and
  http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.sub;hb=HEAD

If the version you run (/home/user/Downloads/bitcoin/db4/db-4.8.30.NC/dist/config.guess) is already up to date, please
send the following data and any information you think might be
pertinent to <config-patches@gnu.org> in order to provide the needed
information to handle your system.

config.guess timestamp = 2009-02-03

uname -m = ppc64le
uname -r = 4.19.0-5-powerpc64le
uname -s = Linux
uname -v = #1 SMP Debian 4.19.37-3 (2019-05-15)

/usr/bin/uname -p = 
/bin/uname -X     = 

hostinfo               = 
/bin/universe          = 
/usr/bin/arch -k       = 
/bin/arch              = 
/usr/bin/oslevel       = 
/usr/convex/getsysinfo = 

UNAME_MACHINE = ppc64le
UNAME_RELEASE = 4.19.0-5-powerpc64le
UNAME_SYSTEM  = Linux
UNAME_VERSION = #1 SMP Debian 4.19.37-3 (2019-05-15)
configure: error: cannot guess build type; you must specify one
~~~

# How reliably can you reproduce the issue, what are the steps to do so?

1. ``./contrib/install_db4.sh `pwd` ``

# What version of Bitcoin Core are you using, where did you get it (website, self-compiled, etc)?

2d1583ee6aff4b68dc46bcb87eb7a85f90b465c3 , self-compiled.

# What type of machine are you observing the error on (OS/CPU and disk type)?

In a VM running Whonix 15 (based on Debian Buster); host is Debian Buster.  ppc64le / POWER9.  HDD.

# Any extra information that might be useful in the debugging process.

The following does work: ``./contrib/install_db4.sh `pwd` --build=powerpc64le``

While it would be nice to fix the configure script to properly detect powerpc64le, I'd settle for a note in `build-unix.md` that specifies the above workaround so that newbies don't have to figure out the configure flag on their own.

--- a/contrib/install_db4.sh
+++ b/contrib/install_db4.sh
@@ -70,6 +70,20 @@ CLANG_CXX11_PATCH_HASH='7a9a47b03fd5fb93a16ef42235fa9512db9b0829cfc3bdf90edd3ec1
 http_get "${CLANG_CXX11_PATCH_URL}" clang.patch "${CLANG_CXX11_PATCH_HASH}"
 patch -p2 < clang.patch
 
+# The packaged config.guess and config.sub are ancient (2009) and can cause build issues.
+# Replace them with modern versions.
+# See https://github.com/bitcoin/bitcoin/issues/16064
+CONFIG_GUESS_URL='https://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.guess;hb=55eaf3e779455c4e5cc9f82efb5278be8f8f900b'
+CONFIG_GUESS_HASH='2d1ff7bca773d2ec3c6217118129220fa72d8adda67c7d2bf79994b3129232c1'
+CONFIG_SUB_URL='https://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.sub;hb=55eaf3e779455c4e5cc9f82efb5278be8f8f900b'
+CONFIG_SUB_HASH='3a4befde9bcdf0fdb2763fc1bfa74e8696df94e1ad7aac8042d133c8ff1d2e32'
+
+rm -f "dist/config.guess"
+rm -f "dist/config.sub"
+
+http_get "${CONFIG_GUESS_URL}" dist/config.guess "${CONFIG_GUESS_HASH}"
+http_get "${CONFIG_SUB_URL}" dist/config.sub "${CONFIG_SUB_HASH}"
+
 cd build_unix/
 
 "${BDB_PREFIX}/${BDB_VERSION}/dist/configure" \

